zimports.py

```diff
@@ -53,13 +53,18 @@ def _rewrite_source(options, filename, source_lines):
 
     # flatten imports into single import per line and rewrite
     # full source
-    if not options.multi_imports:
-        imports = list(
-            _dedupe_single_imports(
-                _as_single_imports(imports, stats, expand_stars=expand_stars),
-                stats,
+    expanded_imports = []
+    for import_node in imports:
+        if options.multi_imports and import_node.is_from:
```

Comment:
it's hard to tell how the behavior changes when multi-imports is set or not, see my groupby approach.